x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -462,15 +462,32 @@ private static void addMetricFields(
         });
     }
 
-    private static void addTimestampField(final DownsampleConfig config, final XContentBuilder builder) throws IOException {
+    private static void addTimestampField(
+        final DownsampleConfig config,
+        Map<String, Object> sourceIndexMappings,
+        final XContentBuilder builder
+    ) throws IOException {
         final String timestampField = config.getTimestampField();
         final String dateIntervalType = config.getIntervalType();
         final String dateInterval = config.getInterval().toString();
         final String timezone = config.getTimeZone();
+        builder.startObject(timestampField);
+
+        MappingVisitor.visitMapping(sourceIndexMappings, (field, mapping) -> {
+            try {
+                if (timestampField.equals(field)) {
+                    final String timestampType = String.valueOf(mapping.get("type"));
+                    builder.field("type", timestampType != null ? timestampType : DateFieldMapper.CONTENT_TYPE);
+                    if (mapping.get("format") != null) {
+                        builder.field("format", mapping.get("format"));
+                    }
+                }
```

Comment:
I am not very sure that the timestamp in the downsample index should have nanosecond precision. Maybe we should just convert it to `date`. On the other hand, keeping the same field type is easy to implement and probably  it would have the least side-effects.

Reply:
Yes I considered that, but I thought that having the same type and format makes more sense. AT least a query running on the original index and the downsampled one would run on the same field type without any difference other than the timestamp granularity. I imagine a situation where a user queries the original index and the downsampled one and I guess they expect the query to work the same way no matter which one of the two indices they query.